lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -638,10 +639,27 @@ class TTIRAllocate : public impl::TTIRAllocateBase<TTIRAllocate> {
 
     module->walk([&](func::FuncOp func) {
       assert(func.getBody().hasOneBlock());
-      SimpleAllocator allocator;
+      auto systemDesc = getCurrentScopeSystemDesc(func);
+      assert(systemDesc);
+      auto addressAlignment = systemDesc.getAddressAlignBytes();
+      SimpleAllocator allocator(addressAlignment);
       Liveness liveness(func.getOperation());
       const LivenessBlockInfo *livenessInfo =
           liveness.getLiveness(&func.getBody().front());
+
+      mlir::SmallVector<Attribute> argumentAllocations;
+      for (auto operand : func.getArguments()) {
+        auto operandTy = mlir::cast<RankedTensorType>(operand.getType());
+        assert(operandTy.getEncoding());
+        auto memorySpace = getMemorySpace(operandTy);
+        auto sizeBytes = getTensorMemrefSizeBytes(operandTy);
+        auto address = allocator.allocate(sizeBytes, memorySpace);
+        argumentAllocations.push_back(rewriter.getAttr<ArgumentAllocationAttr>(
+            address, sizeBytes, memorySpace));
+      }
+      func->setDiscardableAttr("argument_allocations",
```

Comment:
What is a "discardable" attribute?

Reply:
For dialects which haven’t adopted properties yet, the top-level attribute dictionary attached to an operation has special semantics. The attribute entries are considered to be of two different kinds based on whether their dictionary key has a dialect prefix:
inherent attributes are inherent to the definition of an operation’s semantics. The operation itself is expected to verify the consistency of these attributes. An example is the predicate attribute of the arith.cmpi op. These attributes must have names that do not start with a dialect prefix.
discardable attributes have semantics defined externally to the operation itself, but must be compatible with the operations’s semantics. These attributes must have names that start with a dialect prefix. The dialect indicated by the dialect prefix is expected to verify these attributes. An example is the gpu.container_module attribute.
Note that attribute values are allowed to themselves be dictionary attributes, but only the top-level dictionary attribute attached to the operation is subject to the classification above.
When properties are adopted, only discardable attributes are stored in the top-level dictionary, while inherent attributes are stored in the properties storage.

Reply:
I need to update this to have a dialect prefix.